### PR TITLE
lib: pelion: Quick session resume

### PR DESCRIPTION
**Pull request by emob-nordic**
Jira: NCSDK-8953

### CODE REVIEW

File: lib/pelion/Kconfig
Comment (pdunaj) [inline lib/pelion/Kconfig:77]: that cannot be part of choice ;)

**Review by pdunaj — CHANGES_REQUESTED** [review]

File: lib/pelion/Kconfig
Comment (emob-nordic) [reply lib/pelion/Kconfig:77]: Okay. I wanted to make it appear in menuconfig when the option is enabled, seemed strange but looked nice :D I will change that.

Comment (emob-nordic) [reply lib/pelion/Kconfig:79]: done.

Comment (pdunaj) [reply lib/pelion/Kconfig:79]: > wanted to make it appear in menuconfig when the option is enabled

I think you can do that by doing 
```
if PELION_NRF_SECURITY
config ...
  ...
```

Comment (emob-nordic) [reply lib/pelion/Kconfig:79]: I left with `depends on`.

**Comment by emob-nordic** [comment]
Rebased.

**Review by pdunaj — APPROVED** [review]

**Comment by emob-nordic** [comment]
Blocked by: https://github.com/zephyrproject-rtos/zephyr/pull/35501 - must be in downstream so that this PR does not fail doc build.

**Comment by emob-nordic** [comment]
> Blocked by: [zephyrproject-rtos/zephyr#35501](https://github.com/zephyrproject-rtos/zephyr/pull/35501) - must be in downstream so that this PR does not fail doc build.
> Edit: will be included in ongoing upmerge and taken as [nrf fromtree]. Will keep that updated.

Already a part of upmerge PR https://github.com/nrfconnect/sdk-zephyr/pull/518/commits/4408456b393fc534827b85cbbfc8e5ac8009dc6b

**Comment by emob-nordic** [comment]
This PR will be rebased after this is merged: https://github.com/nrfconnect/sdk-nrf/pull/4618

Blocked by: https://github.com/nrfconnect/sdk-nrf/pull/4618

**Comment by rakons** [comment]
The commit description is hard to read. Proposal:

> Add pelion cloud client Kconfig option that enables
> quick client pause -> resume functionality by default.
> This speeds up reconnection by avoiding DTSL handshake.

File: lib/pelion/Kconfig
Comment (rakons) [inline lib/pelion/Kconfig:92]: This option enables quick session resume that avoids DTLS handshake when performing...

**Review by rakons — APPROVED** [review]
Just a comments to the descriptions that are hard to read. No functional comments.

**Comment by emob-nordic** [comment]
> The commit description is hard to read. Proposal:
> 
> > Add pelion cloud client Kconfig option that enables
> > quick client pause -> resume functionality by default.
> > This speeds up reconnection by avoiding DTSL handshake.

That;s right. Reworded as suggested.

**Comment by emob-nordic** [comment]
I accidentally removed the branch... Created #4715 as a replacement.

**Comment by pdunaj** [comment]
@emob-nordic wasn't it easier to simply create the branch again?

**Comment by emob-nordic** [comment]
> @emob-nordic wasn't it easier to simply create the branch again?

Ofc it was, but i didnt notice i can. From now on i will do it this way, but right now lets leave this new pr.

**Pull request closed without merging**
